lms/djangoapps/branding/views.py

```diff
@@ -102,3 +106,20 @@ def courses(request):
     #  we do not expect this case to be reached in cases where
     #  marketing is enabled or the courses are not browsable
     return courseware.views.courses(request)
+
+
+@cache_page(60 * 15)
+def footer(request):
+    # if configuration is not enabled then return 404
+    if not BrandingApiConfig.current().enabled:
+        raise Http404
+    if "application/json" in request.META.get('HTTP_ACCEPT'):
+        return JsonResponse(get_footer(), 200)
+    elif "text/javascript" in request.META.get('HTTP_ACCEPT'):
+        content = get_footer_static("footer.js")
+        return HttpResponse(content, content_type='text/javascript', status=200)
+    elif "text/css" in request.META.get('HTTP_ACCEPT'):
+        content = get_footer_static("footer.css")
+        return HttpResponse(content, content_type='text/javascript', status=200)
+    else:
+        return Http404
```

Comment:
1) It's possible for the client to send "*/*" as the `Accept` header, in which case we should default to `application/json`.
2) If the client doesn't accept any of the formats we send, the HTTP/1.1 standard says we should respond with a 406 (not acceptable) rather than a 404.